scrapy/mail.py

```diff
@@ -5,7 +5,10 @@
 """
 import logging
 
-from six.moves import cStringIO as StringIO
+try:
+    from cStringIO import StringIO as BytesIO
```

Comment:
@redapple you should just remove cStringIO import and just import BytesIO everywhere, to get consistent behavior on Python 2.7 and 3.x

Reply:
what do you mean by "everywhere"?
The other places were ported to Python 3 in a few iterations with a few maintainers. I wouldn't want to change this in elsewhere.
I'll let other maintainers comment and review this PR.

Reply:
io.BytesIO behaves the same on Python 2.7 and Python 3.x.
I see no reason to import cStringIO as BytesIO on Python 2.7.
By "everywhere" I mean Python 2.7 and Python 3.x.

Reply:
@rodrigc cStringIO doesn't copy passed data, while io.BytesIO does a copy, even if access is read-only. This is fixed in Python 3.5, but prior to 3.5 (it seems this includes Python 2.x) io.BytesIO can be 100x slower or more than cStringIO if you just want to provide a file-like interface for existing data. Probably it is not that important for mail sender, but still, using cStringIO when available is a right thing to do.

Reply:
@kmike OK